File: backend/app/schemas/billing.py
"""Pydantic schemas for Billing / Invoices."""
from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class InvoiceCreate(BaseModel):
    sponsor_id: UUID | str
    campaign_id: UUID | str | None = None
    amount_cents: int
    currency: str = "USD"
    due_date: date | None = None
    description: str | None = None


class InvoiceInDB(BaseModel):
    id: UUID | str
    sponsor_id: UUID | str
    sponsor_name: str | None = None
    campaign_id: UUID | str | None = None
    campaign_name: str | None = None
    amount_cents: int
    currency: str
    status: str
    stripe_invoice_id: str | None = None
    stripe_payment_intent_id: str | None = None
    due_date: date | None = None
    paid_at: datetime | None = None
    description: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class BillingSummary(BaseModel):
    total_owed_cents: int
    total_paid_cents: int
    pending_invoices: int


class CheckoutResponse(BaseModel):
    checkout_url: str

File: frontend/src/api/billing.ts
import apiClient from './client';

export interface Invoice {
  id: string;
  sponsor_id: string;
  campaign_id: string | null;
  amount_cents: number;
  currency: string;
  status: string;
  stripe_invoice_id: string | null;
  stripe_payment_intent_id: string | null;
  due_date: string | null;
  paid_at: string | null;
  description: string | null;
  created_at: string;
  updated_at: string;
}

export interface BillingSummary {
  total_owed_cents: number;
  total_paid_cents: number;
  pending_invoices: number;
}

export const listInvoices = async (): Promise<Invoice[]> => {
  const res = await apiClient.get<Invoice[]>('/billing/invoices');
  return res.data;
};

export const getBillingSummary = async (): Promise<BillingSummary> => {
  const res = await apiClient.get<BillingSummary>('/billing/summary');
  return res.data;
};

export const createCheckoutSession = async (invoiceId: string): Promise<{ checkout_url: string }> => {
  const res = await apiClient.post<{ checkout_url: string }>(`/billing/invoices/${invoiceId}/checkout`);
  return res.data;
};
